tests/python_package_test/test_engine.py

```diff
@@ -2887,3 +2888,33 @@ def hook(obj):
     dumped_model_str = str(bst.dump_model(5, 0, object_hook=hook))
     assert "leaf_value" not in dumped_model_str
     assert "LV" in dumped_model_str
+
+
+def test_force_split_with_feature_fraction(tmp_path):
+    X, y = load_boston(return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    lgb_train = lgb.Dataset(X_train, y_train)
+
+    forced_split = {
+        "feature": 0,
+        "threshold": 0.5
+    }
+
+    tmp_split_file = tmp_path / "forced_split.json"
+    with open(tmp_split_file, "w") as f:
+        f.write(json.dumps(forced_split))
+
+    params = {
+        "objective": "regression",
+        "feature_fraction": 0.6,
+        "force_col_wise": True,
+        "feature_fraction_seed": 1,
+        "forcedsplits_filename": tmp_split_file
+    }
+
+    gbm = lgb.train(params, lgb_train)
+    treesDf = gbm.trees_to_dataframe()
```

Comment:
`trees_to_dataframe()` method requires that the `pandas` package should be installed. And `pandas` isn't a hard dependency of the LightGBM Python-package. So, this test should be skipped in case of `pandas` isn't installed.
https://github.com/microsoft/LightGBM/blob/d130bb198b2b9520f0eb6d2a2be3ca1b3bd22c42/tests/python_package_test/test_engine.py#L2505
https://github.com/microsoft/LightGBM/blob/d130bb198b2b9520f0eb6d2a2be3ca1b3bd22c42/python-package/setup.py#L338-L343
#4638

Given that the purpose of this test is to ensure correct behavior of core LightGBM functionality with `feature_fraction` and `forcedsplits_filename` parameters, and `trees_to_dataframe()` method is used only to ease the check of produced results, I suggest to drop the usage of `trees_to_dataframe()` and run this important test unconditionally to the `pandas` installation. This can be achieved by parsing the output of `dump_model()` method directly.

Reply:
Thanks, updated to dump_model()